esmvalcore/_citation.py

```diff
@@ -0,0 +1,229 @@
+    if REFERENCES_PATH and product_tags:
+        # make tags clean and unique
+        tags = list(set(_clean_tags(product_tags)))
+        for tag in tags:
+            citation_entries.append(_collect_bibtex_citation(tag))
+
+    with open(f'{product_name}_citation.bibtex', 'w') as file:
+        file.write('\n'.join(citation_entries))
+
+
+def _clean_tags(tags):
+    """Clean the tags that are recorded as str by provenance."""
+    pattern = re.compile(r'\w+')
+    return pattern.findall(str(tags))
+
+
+def _get_response(url):
+    """Return information from CMIP6 Data Citation service in json format."""
+    json_data = False
+    if url.lower().startswith('https'):
+        try:
+            response = requests.get(url)
+            if response.status_code == 200:
+                json_data = response.json()
+            else:
+                logger.info('Error in the CMIP json link: %s', url)
+        except IOError:
+            logger.info('Error in receiving the CMIP json file')
+    return json_data
+
+
+def _json_to_bibtex(data):
+    """Make a bibtex entry from CMIP6 Data Citation json data."""
+    url = 'url not found'
+    title = data.get('titles', ['title not found'])[0]
+    publisher = data.get('publisher', 'publisher not found')
+    year = data.get('publicationYear', 'publicationYear not found')
+    authors = 'creators not found'
+    doi = 'doi not found'
+
+    author_list = []
+    if data.get('creators', False):
+        author_list = [
+            item.get('creatorName', '') for item in data['creators']
+        ]
+    if author_list:
+        if author_list[0] == author_list[-1]:
+            authors = author_list[0]
+            if not authors:
+                authors = 'creatorName not found'
+        else:
+            authors = ' and '.join(author_list)
+
+    if data.get('identifier', False):
+        doi = data.get('identifier').get('id', 'doi not found')
+        url = f'https://doi.org/{doi}'
+
+    bibtex_entry = (
+        f'{"@misc{"}{url},\n\t'
+        f'url = {{{url}}},\n\t'
+        f'title = {{{title}}},\n\t'
+        f'publisher = {{{publisher}}},\n\t'
+        f'year = {year},\n\t'
+        f'author = {{{authors}}},\n\t'
+        f'doi = {{{doi}}},\n'
+        f'{"}"}\n'
+    )
+    return bibtex_entry
+
+
+def _collect_bibtex_citation(tag):
+    """Collect information from bibtex files."""
+    bibtex_file = REFERENCES_PATH / f'{tag}.bibtex'
+    if bibtex_file.is_file():
+        entry = bibtex_file.read_text()
+    else:
+        logger.info(
+            'The reference file %s does not exist.', bibtex_file
+        )
+        entry = ''
+    return entry
+
+
+def _collect_cmip_citation(json_url):
+    """Collect information from CMIP6 Data Citation Service."""
+    json_data = _get_response(json_url)
+    if json_data:
+        bibtex_entry = _json_to_bibtex(json_data)
+    else:
+        logger.info('Invalid json link %s', json_url)
+        bibtex_entry = False
```

Comment:
```suggestion
        bibtex_entry = ''
```
The convention is to always return a value with the same type from a function. In some cases you may want to use the return value to signal that something went wrong, but then the usual choice is to return `None` instead of `False`. So here it would be best to return either an empty string `''` or `None`.